homeassistant/components/acmeda/cover.py

```diff
@@ -0,0 +1,122 @@
+"""Support for Acmeda Roller Blinds."""
+import asyncio
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    SUPPORT_CLOSE,
+    SUPPORT_CLOSE_TILT,
+    SUPPORT_OPEN,
+    SUPPORT_OPEN_TILT,
+    SUPPORT_SET_POSITION,
+    SUPPORT_SET_TILT_POSITION,
+    SUPPORT_STOP,
+    SUPPORT_STOP_TILT,
+    CoverEntity,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .base import AcmedaBase
+from .const import ACMEDA_HUB_UPDATE, DOMAIN
+from .helpers import add_entities
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Acmeda Rollers from a config entry."""
+    hub = hass.data[DOMAIN][config_entry.entry_id]
+
+    update_lock = asyncio.Lock()
+    current = set()
+
+    async def async_update():
+        async with update_lock:
```

Comment:
I don't think we need the lock. Only one of the jobs can run at any time in the event loop. And we'll move to use a callback.

Reply:
Removed locks.